### PR TITLE
Feature/269 disable post api

### PR DESCRIPTION
**Pull request by uiboy**

### CODE REVIEW

File: openbook_auth/models.py [hunk 1]
```diff
@@ -1445,23 +1445,23 @@ def get_community_post_with_id(self, post_id):
...
-                                          community__memberships__is_moderator=True,
-                                          is_closed=True), Q.OR)
-        post_query_visibility_query.add(Q(community__memberships__user__id=self.pk,
-                                          creator=self,
-                                          is_closed=True), Q.OR)
+        post_query.add(post_query_visibility_query, Q.AND)
 
-        post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, is_closed=False), Q.OR)
+        # Check its not closed or we are community staff
+        post_query_closed_query = Q(is_closed=False)
```
Comment (uiboy) [inline openbook_auth/models.py]: a bit confusing, i assumed `post_query_closed_query` filters out closed posts, but its the opposite. 
perhaps, `post_query_open_query`?

Comment (lifenautjoe) [reply openbook_auth/models.py]: Well it filters out the closed posts if not a staff member 🤔 , anyway this is gone on my next commit where I merged my blocked changes

File: openbook_auth/models.py [hunk 2]
```diff
@@ -1445,23 +1445,23 @@ def get_community_post_with_id(self, post_id):
...
 
-        post_query.add(post_query_visibility_query, Q.AND)
+        post_query_closed_staff_query = Q(community__memberships__user__id=self.pk,
+                                          community__memberships__is_administrator=True) | Q(
+            community__memberships__user__id=self.pk,
+            community__memberships__is_moderator=True)
+
+        post_query_closed_query.add(post_query_closed_staff_query, Q.OR)
```
Comment (uiboy) [inline openbook_auth/models.py]: here we need to check `if_staff` and `is_closed=True` .. this seems to check `is_post_open` OR `is_staff` ..

File: openbook_auth/models.py [hunk 3]
```diff
@@ -1445,23 +1445,23 @@ def get_community_post_with_id(self, post_id):
 
         post_query.add(~Q(community__banned_users__id=self.pk), Q.AND)
 
+        # Check we have membership or its a public community
         post_query_visibility_query = Q(community__memberships__user__id=self.pk)
         post_query_visibility_query.add(Q(community__type=Community.COMMUNITY_TYPE_PUBLIC, ), Q.OR)
-        post_query_visibility_query = Q(community__memberships__user__id=self.pk, is_closed=False)
```
Comment (uiboy) [inline openbook_auth/models.py]: will this not filter all posts where user is member or the community is public ( these should be ANDed with is_closed=False) .. is return _only_ open posts where user is member or community is public

File: openbook_auth/models.py (hunk 2 above)
Comment (lifenautjoe) [reply openbook_auth/models.py]: Ehh the tests pass (?)

Comment (uiboy) [reply openbook_auth/models.py]: yea thats what im wondering the tests are pretty extensive, 🤔

Comment (lifenautjoe) [reply openbook_auth/models.py]: All I did was take your queries and simplify them, not really sure about the logic that should or not happen here.

This checks for the same things your other queries did 🤷‍♂️

File: openbook_auth/models.py (hunk 3 above)
Comment (lifenautjoe) [reply openbook_auth/models.py]: You commented in your own changes 🤔

Comment (lifenautjoe) [reply openbook_auth/models.py]: This line is gone with my changes

Comment (uiboy) [reply openbook_auth/models.py]: yea weird, i commented on ur commit.. but i see it at both places

File: openbook_auth/models.py (hunk 2 above)
Comment (uiboy) [reply openbook_auth/models.py]: Yep, ur right, i maybe missed it while looking at the smaller queries. But the bigger picture adds up 👍

**Comment by lifenautjoe** [comment]
As I'm merging everything I've noted you use positional arguments everywhere. Pls use explicit ones.

So `do_something(user_id=user_id, post_id=post_id)` instead of `do_something(user_id, post_id)`

Makes refactoring waaay easier.

**Pull request merged**